scripts/metric.py

```diff
@@ -6,63 +6,131 @@
+        "PPV Macro": "None",
+        "Kappa 95% CI": "(0.0, 0.0)",
+        ...
+    }
+
+Example plot JSON output:
+    [
+        {"metric": "Overall ACC", "value": 0.85},
+        {"metric": "F1 Macro", "value": "None"},
+        {"metric": "Kappa 95% CI", "value": "(0.0, 0.0)"},
+        ...
+    ]
 
 Functions:
-    calc: Calculate and save performance metrics from prediction output files
+    calculate_all_metrics: Calculate all metrics including confusion matrix and custom metrics
+    calc: Calculate and save performance metrics from prediction output files to metric and plot JSON formats
 """
 
 import argparse
+import json
 from pathlib import Path
 
 import polars as pl
 from pycm import ConfusionMatrix
+from scipy.stats import spearmanr
+
+
+def calculate_all_metrics(
+    actual_values: list,
+    predicted_values: list,
+) -> dict:
+    """Calculate all metrics including confusion matrix metrics and custom metrics.
+
+    This function computes both standard classification metrics via confusion matrix
+    and custom metrics like Spearman correlation. Add new custom metrics here.
+
+    Args:
+        actual_values: List of actual/ground truth values
+        predicted_values: List of predicted values
+
+    Returns:
+        Dictionary of metric names and their values
+    """
+    # Calculate confusion matrix metrics
+    cm = ConfusionMatrix(
+        actual_vector=actual_values,
+        predict_vector=predicted_values,
+    )
+
+    all_metrics = dict(cm.overall_stat.items())
+
+    # Calculate custom metrics
+    # Add your custom metrics below this line
+
+    # Custom metric: Average Spearman correlation
+    spearman_corr, _ = spearmanr(actual_values, predicted_values)
+    all_metrics["Average Spearman"] = spearman_corr
+
+    return all_metrics
 
 
 def calc(
-    output: Path, metric: Path, actual_vector_col: str, predict_vector_col: str
+    prediction: Path,
+    metric: Path,
+    plot: Path,
+    actual_vector_col: str,
+    predict_vector_col: str,
+    selected_metrics: list[str] | None = None,
 ) -> Path:
-    """Calculate performance metrics from prediction output and save to CSV.
+    """Calculate performance metrics from prediction output and save to metric and plot JSON formats.
 
     Reads prediction results from a CSV file, computes classification metrics using
-    a confusion matrix. All metrics are saved to a CSV file.
+    a confusion matrix. All metrics are saved to a JSON file, and a plot-ready version
+    is saved to a separate JSON file for visualization purposes.
 
     Args:
-        output: Path to the CSV file containing prediction results
-        metric: Path where the calculated metrics CSV will be saved
+        prediction: Path to the CSV file containing prediction results
+        metric: Path where the calculated metrics JSON will be saved
+        plot: Path where the plot-ready metrics JSON will be saved
         actual_vector_col: Column name containing actual/ground truth values
```

Comment:
Can we get rid of these and just assume the first column is the actual values and the second is the predicted values?